asmcomp/selectgen.ml

```diff
@@ -672,10 +655,10 @@ method emit_expr (env:environment) exp =
       let r = self#regs_for typ_val in
       Some(self#insert_op env (Iconst_symbol n) [||] r)
   | Cconst_pointer (n, _dbg) ->
-      let r = self#regs_for typ_val in  (* integer as Caml value *)
+      let r = self#regs_for typ_int in
```

Comment:
I think I would have waited before making this change, but at least with the new test we know your patch works

Reply:
I'd prefer to make this change now, as it's more likely to show up bugs this way. Without this change, there could be bugs in this patch that are masked by a conservative typing for Const_pointer.

Reply:
It's only a matter when reviewing your PR, I think; it's better to have this change as part of the PR, but it's the only place that's not necessary for the new `Clet_mut` so it could have been made into a separate commit. I don't advise to change anything though, with the review comments other reviewers should quickly understand why this is here now.